python/ray/air/config.py

```diff
@@ -185,14 +193,20 @@ def _resources_per_worker_not_none(self):
                 # Note that we don't request any CPUs, which avoids possible
                 # scheduling contention. Generally nodes have many more CPUs than
                 # GPUs, so not requesting a CPU does not lead to oversubscription.
-                return {"GPU": 1}
+                resources_per_worker = {"GPU": 1}
             else:
-                return {"CPU": 1}
-        resources_per_worker = {
-            k: v for k, v in self.resources_per_worker.items() if v != 0
-        }
+                resources_per_worker = {"CPU": 1}
+        else:
+            resources_per_worker = {
```

Comment:
nit: I think we can clean up this branching logic a bit now since we don't return early anymore, but we can do it in a separate PR...

Reply:
OK. Let's merge it first. I'll post a followup PR to remove the  branchings like colab & num_workers=None, etc.